modular_framework_core/spawners/external_controllers_spawner.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import roslaunch
+import rospy
+
+# When this node is called, use the roslaunch API to run a node given some parameters saved on the rosparam server
+if __name__ == '__main__':
+    # Initialise node
+    rospy.init_node('launch_grasp_action_server', anonymous=True)
+    # Get information to initialise the node (from rosparam server)
+    package_name = rospy.get_param("package_action_server")
+    node_name = rospy.get_param("node_action_server_name")
+    node_type = rospy.get_param("action_server_type")
+    # If package name is empty it means the server is loaded from a plugin so this node can be shutdown
+    if not package_name:
+        rospy.signal_shutdown("Grasp controller loaded in a plugin, shutting down the node")
```

Comment:
Because in a specific config file, if a given field (package_name) is left empty it means that the user wants to use a ROS controller (loaded with a plugin)